src/modules/sharing/sharing.service.ts

```diff
@@ -1122,18 +1112,16 @@ export class SharingService {
       updatedAt: new Date(),
     });
 
-    await this.removeItemFromBeingShared(
-      dto.itemType,
-      dto.itemId,
-      SharingType.Private,
-    );
-
     const sharing = await this.sharingRepository.findOneSharingBy({
       itemId: dto.itemId,
       itemType: dto.itemType,
       type: SharingType.Public,
     });
 
+    if (!item.isOwnedBy(user) && !sharing) {
```

Comment:
Why ownership is enforced together with the sharing existence? Shouldn't we enforce it always?